### PR TITLE
Remove lmpid and allow passing other ids to secure signals

### PR DESCRIPTION
**Pull request by MO-Thibault**

### CODE REVIEW

File: lib/core/storage.ts [hunk 1]
```diff
@@ -130,5 +102,4 @@ class LocalStorage {
   }
 }
 
-export { LocalStorage, lmpidProvider };
```
Comment (MO-Thibault) [inline lib/core/storage.ts:133]: Not 100% sure I should completely remove this line?

File: lib/addons/gpt.ts [hunk 2]
```diff
@@ -48,23 +48,24 @@ OptableSDK.prototype.installGPTEventListeners = function () {
 };
 
 /*
- * installGPTSecureSignals() sets up loblaw media private ID secure signals on GPT from targeting.
+ * InstallGPTSecureSignals() sets up loblaw media private ID secure signals on GPT from targeting.
+ * Allow for user-defined signal passing. Currently requires specifying the provider name since most signals will be linked to custom ids.
  */
-OptableSDK.prototype.installGPTSecureSignals = function () {
+OptableSDK.prototype.installGPTSecureSignals = function (...signals) {
```
Comment (MO-Thibault) [inline lib/addons/gpt.ts:54]: We will want to add a test for this

File: lib/addons/gpt.ts [hunk 3]
```diff
@@ -48,23 +47,19 @@ OptableSDK.prototype.installGPTEventListeners = function () {
...
+ * InstallGPTSecureSignals() sets up loblaw media private ID secure signals on GPT from targeting.
+ * Allow for user-defined signal passing. Currently requires specifying the provider name since most signals will be linked to custom ids.
  */
-OptableSDK.prototype.installGPTSecureSignals = function () {
-  const sdk = this;
-  sdk.installGPTSecureSignals = function () {};
-
-  window.googletag = window.googletag || { cmd: [], secureSignalProviders: [] };
-  const gpt = window.googletag;
+OptableSDK.prototype.installGPTSecureSignals = function (...signals: [string, string][]) {
```
Comment (MO-Thibault) [inline lib/addons/gpt.ts:53]: We will want a helper function to package signals such as UID2 tokens

File: lib/edge/targeting.ts [hunk 4]
```diff
@@ -43,11 +41,6 @@ async function Targeting(config: Required<OptableConfig>, id: string): Promise<T
   return response;
 }
 
-function LmpidFromCache(config: Required<OptableConfig>): string | null {
-  const ls = new LocalStorage(config);
-  return ls.getLmpid();
-}
-
 function TargetingFromCache(config: Required<OptableConfig>): TargetingResponse | null {
```
Comment (MO-Thibault) [inline lib/edge/targeting.ts:44]: We might want to add an option to set some eids as GAM Secure Signals automatically

**Review by zapo — APPROVED** [review]

File: lib/addons/gpt.ts [hunk 5]
```diff
@@ -48,23 +47,19 @@ OptableSDK.prototype.installGPTEventListeners = function () {
...
-  if (lmpid) {
-    gpt.cmd.push(function () {
-      gpt.secureSignalProviders.push({
-        id: lmpidProvider,
-        collectorFunction: function () {
-          return Promise.resolve(lmpid);
-        },
+  if (signals && signals.length > 0) {
+    signals.forEach(({ provider, id }) => {
+      gpt.cmd.push(() => {
```
Comment (zapo) [inline lib/addons/gpt.ts:58]: Can we inverse those so that only a single function is pushed to gpt?

**Review by zapo — APPROVED** [review]

**Pull request merged**